### ISSUE TITLE
[Bug]: Random colors/opacity applies only one value to all particles when using SVG shape

### ISSUE THREAD

**Issue by Jaessun** [tsparticles/tsparticles#4532]
### Contact Details

jason@octopix.be

### What happened?

Hi,

I've provided an array of hex codes for my particles colors. When using a built-in shape type (e.g. `circle`), each particle picks a random color from the provided array of colors, which is the expected behaviour.

However when using an SVG image shape with `"replaceColor": true`, a single random color is picked and then used for every particle on the screen, instead of each particle individually picking a random color.

The same kind of behaviour happens with a random opacity, where a single opacity value will be picked at random then used for all the particles.

Please use the bellow configuration to reproduce the issue. I believe switching the value of `particles.shape.type` between `image` and `circle` should make the issue pretty clear.

### tsParticles Version

2.1.3

### Which library are you using?

Vanilla (tsparticles)

### tsParticles Configuration

```json
{
   "autoPlay":true,
   "fullScreen":{
      "enable":true
   },
   "particles":{
      "color":{
         "value":[
            "#ff0000",
            "#00ff00",
            "#0000ff"
         ]
      },
      "move":{
         "direction":"bottom",
         "enable":true,
         "speed":{
            "min":1,
            "max":5
         }
      },
      "number":{
         "value":100
      },
      "opacity":{
         "random":{
            "enable":true,
            "minimumValue":0.25
         },
         "value":{
            "min":0.25,
            "max":1
         }
      },
      "shape":{
         "options":{
            "image":{
               "src":"https://upload.wikimedia.org/wikipedia/fr/c/c8/Twitter_Bird.svg",
               "width":32,
               "height":32,
               "replaceColor":true
            }
         },
         "type":"image"
      },
      "size":{
         "random":{
            "enable":true,
            "minimumValue":1
         },
         "value":{
            "min":5,
            "max":10
         }
      }
   }
}
```

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by matteobruni** [comment]
Hi @Jaessun,

This was discussed on Discord a few days ago. It's not a bug, it's intended to work this way since every color replace is a new image to load. To avoid loading more images (this could involve `"random"` value as well), it will be used only one color.

You can still force it copying the image multiple times and assigning to each image a different color. You can see a sample config below
 https://github.com/matteobruni/tsparticles/blob/e08d09f7243c4fd790b990e1c3f75c832b9e3ef3/demo/vanilla/public/presets/svgReplace.json#L115-L143

**Comment by Jaessun** [comment]
Thanks for the quick response and for the example, I think that should solve the problem for the colors.

But what about the opacity, is that also intended? There are too many possible values between min and max to copy the image for each value. Is there a solution for that?

**Comment by matteobruni** [comment]
Opacity shouldn't be affected, I didn't notice the issue about opacity. I'll do some checks and I'll let you know

**Comment by matteobruni** [comment]
Yes, opacity were ignored only for svg replaced images, it will be released in the upcoming 2.2.0 (PR #4341)
